src/components/MAPI/releases/ClusterDetailWidgetRelease.tsx

```diff
@@ -134,61 +159,150 @@ const ClusterDetailWidgetRelease: React.FC<IClusterDetailWidgetReleaseProps> = (
     ? releasev1alpha1.getReleaseNotesURL(currentRelease)
     : undefined;
 
+  const [targetVersion, setTargetVersion] = useState('');
+  const targetRelease = useMemo(() => {
+    if (!releaseList) return undefined;
+
+    return releaseList.items.find(
+      (v) => v.metadata.name.slice(1) === targetVersion
+    );
+  }, [releaseList, targetVersion]);
+
+  const [upgradeModalVisible, setUpgradeModalVisible] = useState(false);
+
+  const handleUpgradeModalClose = () => {
+    setUpgradeModalVisible(false);
+
+    /**
+     * Reset target version only after modal closes,
+     * to prevent flashes in the modal's content
+     */
+    setTimeout(() => {
```

Comment:
I don't really like this, but without doing it, in the few ms that it takes for the modal to animate out, its contents would change, and it would look weird.